server/tests/classes.test.ts

```diff
@@ -0,0 +1,518 @@
+
+            expect(response.status).toBe(200)
+            expect(response.body.title).toBe('Clase 1 actualizada')
+        })
+
+        it('Invalid field', async () => {
+            findByIdClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+
+            createClassMock.mockImplementation(async () => {
+                throw new Error('Invalid data')
+            })
+
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1')
+                .field('description', 1)
+                .attach(
+                    'file',
+                    // TODO: Change to real video file
+                    Buffer.from('test file content', 'utf-8'),
+                    'mockedFile1.txt'
+                )
+
+            expect(response.status).toBe(400)
+            expect(response.body.error).toBe('Invalid file type. Only quicktime,mp4 and mpeg video files are allowed.')
+        })
+
+        it('No fields to update provided', async () => {
+            findByIdClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+
+            createClassMock.mockImplementation(async () => {
+                throw new Error('No fields to update provided')
+            })
+
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+
+            expect(response.status).toBe(400)
+            expect(response.body.error).toBe('No fields to update provided')
+        })
+
+        it('Invalid action, same fields without file provided', async () => {
+            findByIdClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+
+            createClassMock.mockImplementation(async () => {
+                throw new Error('Invalid data')
+            })
+
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1')
+                .field('description', 1)
+                .attach(
+                    'file',
+                    Buffer.from('test file content', 'utf-8'),
+                    'mockedFile1.txt'
+                )
+
+            expect(response.status).toBe(400)
+            expect(response.body.error).toBe('Invalid file type. Only quicktime,mp4 and mpeg video files are allowed.')
+        })
+
+        it('Should return unauthenticated error', async () => {
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .field('title', 'Clase 1')
+                .field('description', 'Descripción 1')
+                .attach(
+                    'file',
+                    // TODO: Change to real video file
```

Comment:
Delete this comment please. I understand that 'EVERYTHING' has already been done, although it is still better to put an mp4 file in the buffer, for code coherence